### PR TITLE
Added tables to database for country code and country sub division look-ups

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
fixes https://github.com/mitodl/micromasters/issues/2990

#### What's this PR do?
- Creates tables to add country and their sub division info

#### How should this be manually tested?
- use django shell docker-compose run web ./manage.py shell
- and browse objects 

@pdpinch

### CODE REVIEW

**Comment by noisecapella** [comment]
There are some countries in `iso-3166-2` which are not in the CSV list and vice versa. Can you update the CSV list to match the iso-3166-2 library list?

>>> amir_country.difference(isocountry)
{'XK', 'GZ', 'AN'}
>>> isocountry.difference(amir_country
... )
{'UM', 'BL', 'CW', 'SX', 'SS', 'AX', 'BQ', 'MF'}

**Comment by noisecapella** [comment]
@pdpinch @Ferdi I'm not sure we should add the latitude/longitude pairs in this PR. We should use a `PointField` for those, but in any case it would be clearer if we had an idea what kind of queries we want. If we want to look up what country a user is in we can populate a database with region shapes and use postgres's GIS features to tell if a lat/lon pair is inside that country. As it is I don't think these positions will be useful to users in redash, it's just some arbitrary location in the center of the country.

**Comment by Ferdi** [comment]
@noisecapella  The common use case is "show a visualization of DEDP users by country".  Given that we have the country for each users, a simple lat/long will be enough to associate users with countries in the map. So lat/long are associated with the country , not the user.  Does that make sense ?

@pdpinch are you sure we need sub/divisions  ?

**Comment by pdpinch** [comment]
As far as I know, we don't have a immediate use case for subdivision. 

At some point in the future, it would be nice to provide the list of subdivisions by API instead of requiring all users to download the json for all subdivisions, but I don't think we're going to pursue that optimization soon. 

I also learned at lunch that we already have subdivisions (and countries) in the pycountry package, which was added for Pearson integration. The only thing missing from there is the lat/long data.  

I'm fine with dropping the subdivision table. 

> @pdpinch <https://github.com/pdpinch> are you sure we need sub/divisions ?

**Comment by noisecapella** [comment]
Ok, in that case @amir-qayyum-khan can you look into using a [PointField](https://docs.djangoproject.com/en/1.10/ref/contrib/gis/model-api/#pointfield) to store the position data?

**Comment by amir-qayyum-khan** [comment]
@noisecapella i am getting as issue with PointField  and sqlite https://github.com/coderholic/django-cities/issues/6. They suggest to use SpatialLite
http://stackoverflow.com/questions/12538510/getting-databaseoperations-object-has-no-attribute-geo-db-type-error-when-do?rq=1

```python
Applying profiles.0030_country...Traceback (most recent call last):
  File "./manage.py", line 14, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 359, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/base.py", line 294, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/base.py", line 345, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/commands/migrate.py", line 204, in handle
    fake_initial=fake_initial,
  File "/usr/local/lib/python3.5/site-packages/django/db/migrations/executor.py", line 115, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.5/site-packages/django/db/migrations/executor.py", line 145, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.5/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.5/site-packages/django/db/migrations/migration.py", line 129, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python3.5/site-packages/django/db/migrations/operations/models.py", line 96, in database_forwards
    schema_editor.create_model(model)
  File "/usr/local/lib/python3.5/site-packages/django/db/backends/base/schema.py", line 246, in create_model
    definition, extra_params = self.column_sql(model, field)
  File "/usr/local/lib/python3.5/site-packages/django/db/backends/base/schema.py", line 136, in column_sql
    db_params = field.db_parameters(connection=self.connection)
  File "/usr/local/lib/python3.5/site-packages/django/db/models/fields/__init__.py", line 647, in db_parameters
    type_string = self.db_type(connection)
  File "/usr/local/lib/python3.5/site-packages/django/contrib/gis/db/models/fields.py", line 124, in db_type
    return connection.ops.geo_db_type(self)
AttributeError: 'DatabaseOperations' object has no attribute 'geo_db_type'
```

**Comment by amir-qayyum-khan** [comment]
should i go for SpatialLite?

**Comment by noisecapella** [comment]
@amir-qayyum-khan We aren't using sqlite so you should try this answer instead to set the engine to postgis: http://stackoverflow.com/questions/12538510/getting-databaseoperations-object-has-no-attribute-geo-db-type-error-when-do?rq=1#answer-21317596

**Comment by amir-qayyum-khan** [comment]
@noisecapella the i gives 
```python
Traceback (most recent call last):
  File "./manage.py", line 14, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 341, in execute
    django.setup()
  File "/usr/local/lib/python3.5/site-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.5/site-packages/django/apps/registry.py", line 108, in populate
    app_config.import_models(all_models)
  File "/usr/local/lib/python3.5/site-packages/django/apps/config.py", line 199, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/local/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 673, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "/usr/local/lib/python3.5/site-packages/django/contrib/auth/models.py", line 4, in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
  File "/usr/local/lib/python3.5/site-packages/django/contrib/auth/base_user.py", line 52, in <module>
    class AbstractBaseUser(models.Model):
  File "/usr/local/lib/python3.5/site-packages/django/db/models/base.py", line 119, in __new__
    new_class.add_to_class('_meta', Options(meta, app_label))
  File "/usr/local/lib/python3.5/site-packages/django/db/models/base.py", line 316, in add_to_class
    value.contribute_to_class(cls, name)
  File "/usr/local/lib/python3.5/site-packages/django/db/models/options.py", line 214, in contribute_to_class
    self.db_table = truncate_name(self.db_table, connection.ops.max_name_length())
  File "/usr/local/lib/python3.5/site-packages/django/db/__init__.py", line 33, in __getattr__
    return getattr(connections[DEFAULT_DB_ALIAS], item)
  File "/usr/local/lib/python3.5/site-packages/django/db/utils.py", line 211, in __getitem__
    backend = load_backend(db['ENGINE'])
  File "/usr/local/lib/python3.5/site-packages/django/db/utils.py", line 134, in load_backend
    raise ImproperlyConfigured(error_msg)
django.core.exceptions.ImproperlyConfigured: 'django.contrib.gis.db.backends.postgis' isn't an available database backend.
Try using 'django.db.backends.XXX', where XXX is one of:
    'mysql', 'oracle', 'postgresql', 'sqlite3'
Error was: cannot import name 'GDALRaster'
```

**Comment by amir-qayyum-khan** [comment]
Done @noisecapella 

https://github.com/mitodl/micromasters/pull/3032#issuecomment-293375885

File: RELEASE.rst [hunk 1]
```diff
@@ -1,6 +1,17 @@
 Release Notes
 =============
 
+Version 0.54.0
```
Comment (noisecapella) [inline RELEASE.rst]: Can you remove this from the PR?

File: micromasters/settings.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 from django.core.exceptions import ImproperlyConfigured
 
 
-VERSION = "0.53.12"
+VERSION = "0.54.0"
```
Comment (noisecapella) [inline micromasters/settings.py]: This should be reverted

File: profiles/migrations/0034_data_migration_for_countries.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+from profiles.models import Country
+
+
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+def forwards(apps, schema_editor):
+    """load list of countries from file"""
+    countries_file = 'data/countries.csv'
+    countries_file_path = '{}/{}'.format(BASE_DIR, countries_file)
```
Comment (noisecapella) [inline profiles/migrations/0034_data_migration_for_countries.py]: Please use `os.path.join` here

**Comment by amir-qayyum-khan** [comment]
Done @noisecapella

**Comment by amir-qayyum-khan** [comment]
Done @noisecapella

File: profiles/migrations/0034_data_migration_for_countries.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+from profiles.models import Country
+
+
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+def forwards(apps, schema_editor):
+    """load list of countries from file"""
+    countries_file = 'data/countries.csv'
+    countries_file_path = os.path.join(os.sep, BASE_DIR, countries_file)
```
Comment (noisecapella) [inline profiles/migrations/0034_data_migration_for_countries.py]: Can you remove `os.sep` here? It shouldn't be necessary, `BASE_DIR` is an absolute path by itself

**Comment by amir-qayyum-khan** [comment]
done @noisecapella

**Pull request merged**